stac_check/lint.py

```diff
@@ -18,15 +17,16 @@ class Linter:
     config_file: Optional[str] = None
     assets: bool = False
     links: bool = False
-    recursive: bool = False
+    recursive: int = 0
```

Comment:
In `stac-validator` we separated the recursion flag from max depth, e.g.: https://github.com/stac-utils/stac-validator/blob/f8647f8ce3ac44e355d68bdf3f17cfc9914f689c/stac_validator/stac_validator.py#L31-L42. IMO we should use the same pattern here?

Reply:
true